src/local-ssh/sshTunnel.ts

```diff
@@ -45,7 +45,7 @@ export class SupervisorSSHTunnel {
 	constructor(
 		private readonly logger: ILogService,
 		readonly workspaceInfo: GetWorkspaceAuthInfoResponse,
-		private readonly localsshService: LocalSSHServiceImpl,
+		private extensionIpc: Client<ExtensionServiceDefinition>,
 	) { }
 
 	private createPrivateKey(): Promise<string> {
```

Comment:
Let's move this to the extension side, so the client ask for owner token and sshkey at the same time

Reply:
I don't understand, we use extensionIpc here to report error only

sshkey is what we ask supervisor to create, we also need to tunnel supervisor here. I'm not going to change it

Reply:
This github comment UI is not clear :facepalm: , I'm referring to `createPrivateKey`

Reply:
Yes, I mean they are doing the same functionality - tunnel ssh, extension don't need to know how we can tunnel to supervisor.

Reply:
> extension don't need to know how we can tunnel to supervisor or even know that we are using tunnel or ssh gateway 

not sure I'm following it will just call the supervisor api, so proxy.js get all required credentials to authenticate and not do any side request that has to be handled and then reported back to extension